### PR TITLE
Move sync code to pkg/skaffold/sync/kubectl

### LINKED ISSUE #1136: move pkg/skaffold/kubernetes/sync* to pkg/skaffold/sync/kubectl
That is the correct place for the code

### PR DESCRIPTION
**Pull request by priyawadhwa**
This is the correct place for this code. Might require some
reorganization if we add other syncers, like rsync, but that shouldn't
be difficult.

Should fix #1136

### CODE REVIEW

**Comment by r2d4** [comment]
and the linter issue

pkg/skaffold/sync/kubectl/kubectl.go:34:6: type name will be used as kubectl.KubectlSyncer by other packages, and that stutters; consider calling this Syncer (golint)

File: pkg/skaffold/sync/sync.go [hunk 1]
```diff
@@ -108,3 +112,44 @@ func intersect(context string, syncMap map[string]string, files []string) (map[s
 	}
 	return ret, nil
 }
+
+func Perform(ctx context.Context, image string, files map[string]string, cmdFn func(context.Context, v1.Pod, v1.Container, string, string) *exec.Cmd) error {
```
Comment (r2d4) [inline pkg/skaffold/sync/sync.go]: You can still leave this as unexported, I dont think we want it used outside the package.

Comment (priyawadhwa) [reply pkg/skaffold/sync/sync.go]: so the `Sync` function in the kubectl package calls it, so I think it has to be exported?

Comment (r2d4) [reply pkg/skaffold/sync/sync.go]: doh you are correct, LGTM then

**Pull request merged**